src/FSharpAux/Array.fs

```diff
@@ -226,6 +228,106 @@ module Array =
                 | _               -> loop arr (currentIdx+stepSize) 
         loop arr startIdx 
 
+    /// Returns a new array containing only the elements of the input array for which the given predicate returns true.
+    let filteri (predicate : int -> 'T -> bool) (array : 'T []) =
+        let mutable i = -1
+        Array.filter (
+            fun x ->
+                i <- i + 1
+                predicate i x
+        ) array
+
+    /// Returns the length of an array containing only the elements of the input array for which the given predicate returns true.
+    let filterLength (predicate : 'T -> bool) (array : 'T []) =
```

Comment:
Maybe change name to something more in Line with other F# function names.
E.g.  "countIf", "countByPredicate"